### ISSUE TITLE
Migrate to Route Handlers from API Routes

### ISSUE THREAD

**Issue by aryanprince** [UniTrackApp/dashboard#83]
## Summary

Basically switching from `pages/api` to `app/api`. They do the same thing, but the new app directory is much better (see below why).

## Why

- There's more features
- Better web standards being maintained in the new `app` router (app router now uses native Web Request APIs instead of using Next.js properietery APIs in prev `pages` router)
- Easy to set up different REST API endpoints just by using functions with verbs as names. Need a `GET` endpoint? Create a `GET()` function. Need a `POST/PUT/DELETE` endpoint? Create a function with that name, that's it.

Basically the `app` router is the future of Next.js and it would be awesome if we could migrate. 

Shouldn't be that complex of a migration, compared to the nightmare of switching the 'frontend' code to `app` router, remember that big PR I made few weeks ago? 😂

## Links to Docs

Our existing API Routes: https://nextjs.org/docs/pages/building-your-application/routing/api-routes

New API Route Handlers: https://nextjs.org/docs/app/building-your-application/routing/route-handlers
